exchanges/kraken/kraken_websocket.go

```diff
@@ -550,74 +550,81 @@ func (k *Kraken) wsProcessOpenOrders(ownOrders interface{}) error {
 				return err
 			}
 			for key, val := range result {
-				var oStatus order.Status
-				oStatus, err = order.StringToOrderStatus(val.Status)
-				if err != nil {
-					k.Websocket.DataHandler <- order.ClassificationError{
-						Exchange: k.Name,
-						OrderID:  key,
-						Err:      err,
-					}
+				d := &order.Detail{
+					Exchange:             k.Name,
+					OrderID:              key,
+					AverageExecutedPrice: val.AveragePrice,
+					Amount:               val.Volume,
+					LimitPriceUpper:      val.LimitPrice,
+					ExecutedAmount:       val.ExecutedVolume,
+					Fee:                  val.Fee,
+					Date:                 convert.TimeFromUnixTimestampDecimal(val.OpenTime).Truncate(time.Microsecond),
+					LastUpdated:          convert.TimeFromUnixTimestampDecimal(val.LastUpdated).Truncate(time.Microsecond),
 				}
-				if val.Description.Price > 0 {
-					oSide, err := order.StringToOrderSide(val.Description.Type)
-					if err != nil {
+
+				if val.Status != "" {
+					if s, err := order.StringToOrderStatus(val.Status); err != nil {
 						k.Websocket.DataHandler <- order.ClassificationError{
 							Exchange: k.Name,
 							OrderID:  key,
 							Err:      err,
 						}
+					} else {
+						d.Status = s
 					}
+				}
+
+				if val.Description.Pair != "" {
 					if strings.Contains(val.Description.Order, "sell") {
-						oSide = order.Sell
+						d.Side = order.Sell
+					} else {
+						if oSide, err := order.StringToOrderSide(val.Description.Type); err != nil {
+							k.Websocket.DataHandler <- order.ClassificationError{
+								Exchange: k.Name,
+								OrderID:  key,
+								Err:      err,
+							}
+						} else {
+							d.Side = oSide
+						}
 					}
-					oType, err := order.StringToOrderType(val.Description.OrderType)
-					if err != nil {
+
+					if oType, err := order.StringToOrderType(val.Description.OrderType); err != nil {
 						k.Websocket.DataHandler <- order.ClassificationError{
 							Exchange: k.Name,
 							OrderID:  key,
 							Err:      err,
 						}
+					} else {
+						d.Type = oType
 					}
 
-					p, err := currency.NewPairFromString(val.Description.Pair)
-					if err != nil {
+					if p, err := currency.NewPairFromString(val.Description.Pair); err != nil {
 						k.Websocket.DataHandler <- order.ClassificationError{
 							Exchange: k.Name,
 							OrderID:  key,
 							Err:      err,
 						}
+					} else {
+						d.Pair = p
+						if a, err := k.GetPairAssetType(p); err != nil {
+							return err
```

Comment:
Since I'm already requesting a change, I think I'd like this to be an order classification error instead of returning an error. For example, if multiple updates come through, then all the other updates are dropped. Plus the order manager does not require an asset for an order to be added into the system

Reply:
Agreed.